app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -47,6 +50,13 @@ class MetaRPCClient {
   close() {
     this.notificationChannel.removeAllListeners();
     this.uncaughtErrorChannel.removeAllListeners();
+    // fail all unfinished requests
+    for (const [id, handler] of this.requests) {
+      if (!this.responseHandled[id]) {
+        this.responseHandled[id] = true;
+        handler(new DisconnectError('disconnected'));
+      }
+    }
```

Comment:
If connection closes while requests are in progress they will receive exception.